Lambda calculator implementation

DanaKhromenko opened the pull request:
- added interface Calculator with method "calculate";
- implemented Calculator with class MyCalculator;
- added class for MyCalculator testing named MyCalculatorTest.

DanaKhromenko commented:
- changed imports order;
- separated "import static org.junit.jupiter.api.Assertions.*" from "import org.junit.jupiter.api.Test" with extra line.

RomanDubovskyi commented on src/main/java/core/basesyntax/MyCalculator.java:
```diff
@@ -0,0 +1,24 @@
+            case '^':
+                return Math.pow(operand1, operand2);
+            default:
+                throw new RuntimeException("Unknown operation " + operand1);
```
```suggestion
                throw new RuntimeException("Unknown operation " + operation);
```

RomanDubovskyi commented on src/test/java/core/basesyntax/MyCalculatorTest.java:
```diff
@@ -0,0 +1,1275 @@
+import org.junit.jupiter.api.Test;
+
+public class MyCalculatorTest {
+    private static final MyCalculator CALCULATOR = new MyCalculator();
```
```suggestion
    private static final MyCalculator calculator = new MyCalculator();
```
не immutable класи ловеркейсом пишуться

RomanDubovskyi commented on src/test/java/core/basesyntax/MyCalculatorTest.java:
```diff
@@ -0,0 +1,1275 @@
+
+    // 1. ADDITION
+    @Test
+    void addictionPositiveOperands_Ok() {
```
```suggestion
    void additionPositiveOperands_Ok() {
```

RomanDubovskyi commented on src/main/java/core/basesyntax/Calculator.java:
```diff
@@ -0,0 +1,5 @@
+package core.basesyntax;
+
+public interface Calculator {
+    public double calculate(double operand1, double operand2, char operation);
```
```suggestion
    double calculate(double operand1, double operand2, char operation);
```
у інтерфейся методи по дефолту є публічні і абстрактні

RomanDubovskyi commented on src/test/java/core/basesyntax/MyCalculatorTest.java:
```diff
@@ -0,0 +1,1275 @@
+    }
+
+    @Test
+    void addictionPositiveOperands_NotOk() {
```
фактично роботу цього тесту в себе включає попередній, у нас не може бути ситуації коли один пройшов а інший ні. Аналогічно з іншими notOk випадками крім тих, де ми власне оічкуємо якийсь конкретний ексепшин. Тоді це доцільно, бо ми хочемо подивитись, що там випадає ікспектед ексепшин а не NPE наприклад

DanaKhromenko commented:
- edited exception's message;
- changed variable's name from CALCULATOR to calculator;
- replaced "addiction" with "addition" in all methods that are testing addition;
- removed public modifier from interface;
- removed *_NotOk methods, where couldn't catch an exception.
